### ISSUE TITLE
OB1 Collector long WakeUp

### ISSUE THREAD

**Issue by thorinEichenschild007** [NightscoutFoundation/xDrip#2111]
Hello,
I Updated my OnePlus 8T to the OxygenOS 12,
and since the Update, the OB1 Collector has a very long WakeUp Time.
So there are 3 Readings, one missing, and Three Readings again.

Maybe you could add an option to restart the OB1 30seconds before the next reading?

If I restart it manually, it's working.

I think, there are some changes in the Android Kernel, in OxygenOS 11 I had no problems with the collector service.

**Comment by jamorham** [comment]
Check what power saving options are present in this version of Android and exclude xDrip from them. Make sure it is allowed to work in the background.

**Comment by thorinEichenschild007** [comment]
I excluded it from all settings I found.
I think, there are Settings in the Kernel which I cannot change.
I also allowed the background activity for xDrip Plus.

**Comment by thorinEichenschild007** [comment]
That's the Problem: 
![Screenshot_2022-05-10-14-48-47-83_2dd2c1bcb8f659cd696bcba2072f751b](https://user-images.githubusercontent.com/91971937/167632108-2a18a9f3-c202-464d-9731-52a91417466a.jpg)

**Comment by thorinEichenschild007** [comment]
I found one point in the Code which looked like a solution in
/app/src/main/java/com/eveningoutpost/dexdrip/SystemStatus.java

I'm no expert in Java, maybe you can add the circled point into a time based loop? 
I don't trust me to try it out, because I don't want to crash my phone, and I cannot build a APK file. 

<img width="497" alt="Screenshot 2022-05-10 164430" src="https://user-images.githubusercontent.com/91971937/167656945-f1aff3e9-f05d-4341-8965-b6476b4218d3.png">

**Comment by tolot27** [comment]
Since Dexcom only communicates every 5 minutes and your last connection is 1 second ago, there is no need to wake up earlier than these 4 minutes.

**Comment by thorinEichenschild007** [comment]
The OB1 Collector starts to scan Around 30seconds before next reading. 
Sometimes, the collector needs 4minutes to start, that's why the readings are Sometimes missing

**Comment by Navid200** [comment]
I don't think it makes any difference how much earlier we wake up the scanner if the cause of the problem is that Android doesn't allow xDrip to operate in the background.
You may be looking at the effect rather than the cause.

**Comment by thorinEichenschild007** [comment]
Do you know an app which can force the Background Activity for other apps?

**Comment by Navid200** [comment]
Not sure why you say that.  I just checked Android 12 in an emulator and can see it.

**Comment by thorinEichenschild007** [comment]
There is no option for the Battery optimization for Bluetooth in Android 12

**Comment by Navid200** [comment]
I already told you what I did to verify.
Open a discussion and I can show you.  
Or, ask many others who have been using Android 12 and xDrip.

**Comment by thorinEichenschild007** [comment]
Maybe it's a bug in OxygenOS 12.
